File: src/app/Class/Comment.ts
import { Users } from './Users';
import { Song } from './Song';

export class Comment {

    private id: number;
    private user: Users;
    private song: Song;
    private commentDate: Date;
    private wasEdited: boolean;
    private commentContent: string;

    constructor(obj?: any) {
        this.id = (obj && obj.id) || 0;
        this.user = (obj && obj.user) || null;
        this.song = (obj && obj.song) || null;
        this.commentDate = (obj && obj.commentDate) || null;
        this.wasEdited = (obj && obj.wasEdited) || false;
        this.commentContent = (obj && obj.commentContent) || '';
    }

    // Getters

    getId(): number {
        return this.id;
    }

    getUser(): Users {
        return this.user;
    }

    getSong(): Song {
        return this.song;
    }

    getCommentDate(): Date {
        return this.commentDate;
    }

    getWasEdited(): boolean {
        return this.wasEdited;
    }

    getCommentContent(): string {
        return this.commentContent;
    }


    // Setters
    
    setUser(user: Users) {
        this.user = user;
    }

    getSong(song: Song){
        this.song = song;
    }

    getCommentDate(commentDate: Date) {
        this.commentDate = commentDate;
    }

    getWasEdited(wasEdited: boolean) {
        this.wasEdited = wasEdited;
    }

    getCommentContent(commentContent: string) {
        this.commentContent = commentContent;
    }

}

File: src/main/java/musiccube/entities/Comment.java
package musiccube.entities;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "comment")
public class Comment {

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(referencedColumnName = "id")
    private User user;

    @ManyToOne
    @JoinColumn(referencedColumnName = "id")
    private Song song;

    @ManyToOne
    @JoinColumn(referencedColumnName = "id")
    private Album album;

    @ManyToOne
    @JoinColumn(referencedColumnName = "id")
    private Band band;

    @ManyToOne
    @JoinColumn(name = "artist", referencedColumnName = "_id")
    private Artist artist;

    @Column
    private Date commentDate;

    @Column
    private boolean wasEdited;

    @Column
    private String commentContent;

    public Comment() {
    }

    private Comment(User user, Date commentDate, boolean wasEdited, String commentContent) {
        this.user = user;
        this.commentDate = commentDate;
        this.wasEdited = wasEdited;
        this.commentContent = commentContent;
    }

    public Comment(User user, Song song, Date commentDate, boolean wasEdited, String commentContent) {
        this(user, commentDate, wasEdited, commentContent);
        this.song = song;
    }

    public Comment(User user, Album album, Date commentDate, boolean wasEdited, String commentContent) {
        this(user, commentDate, wasEdited, commentContent);
        this.album = album;
    }

    public Comment(User user, Band band, Date commentDate, boolean wasEdited, String commentContent) {
        this(user, commentDate, wasEdited, commentContent);
        this.band = band;
    }

    public Comment(User user, Artist artist, Date commentDate, boolean wasEdited, String commentContent) {
        this(user, commentDate, wasEdited, commentContent);
        this.artist = artist;
    }

    public int getId() {
        return id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Song getSong() {
        return song;
    }

    public void setSong(Song song) {
        this.song = song;
    }

    public Album getAlbum() {
        return album;
    }

    public void setAlbum(Album album) {
        this.album = album;
    }

    public Band getBand() {
        return band;
    }

    public void setBand(Band band) {
        this.band = band;
    }

    public Artist getArtist() {
        return artist;
    }

    public void setArtist(Artist artist) {
        this.artist = artist;
    }

    public Date getCommentDate() {
        return commentDate;
    }

    public void setCommentDate(Date commentDate) {
        this.commentDate = commentDate;
    }

    public boolean isWasEdited() {
        return wasEdited;
    }

    public void setWasEdited(boolean wasEdited) {
        this.wasEdited = wasEdited;
    }

    public String getCommentContent() {
        return commentContent;
    }

    public void setCommentContent(String commentContent) {
        this.commentContent = commentContent;
    }

}
